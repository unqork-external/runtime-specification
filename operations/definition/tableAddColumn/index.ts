export * from './tableAddColumn.model'
export * from './tableAddColumn.options'
