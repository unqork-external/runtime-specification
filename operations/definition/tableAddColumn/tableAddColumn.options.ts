import { CollectionOf, Description, Property } from '@tsed/schema'

import { BaseComponentDefinition } from '../../../base-component-interface'
export class TableAddColumnOperationOptions {
  @Description(
    'A unique string identifier used to specify the particular table ' +
      'or component in the application where the column addition operation is to be executed',
  )
  @Property()
  targetKey: string

  @Description('The component definition to be added to the target table ')
  @Property()
  @CollectionOf(() => BaseComponentDefinition)
  definitions: BaseComponentDefinition[]

  @Description('Specifies whether the component is a Vega component')
  @Property()
  isVega: boolean
}
