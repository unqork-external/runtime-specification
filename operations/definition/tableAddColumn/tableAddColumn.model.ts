import { Description, DiscriminatorValue, Required } from '@tsed/schema'

import { TableAddColumnOperationOptions } from './tableAddColumn.options'
import { OpsBuilderStable, Stable } from '../../../../decorators/stability'
import { OperationTypes } from '../../enums/operation-types.enum'
import { Operation } from '../../interface/operations.interface'

@DiscriminatorValue(OperationTypes.TABLE_ADD_COLUMN)
@Description('Adds columns to the Table component based on provided component definition(s).')
@Stable()
@OpsBuilderStable()
export class TableAddColumnOperation extends Operation<OperationTypes.TABLE_ADD_COLUMN> {
  public type = OperationTypes.TABLE_ADD_COLUMN as const

  @Required()
  public options: TableAddColumnOperationOptions = new TableAddColumnOperationOptions()
}
