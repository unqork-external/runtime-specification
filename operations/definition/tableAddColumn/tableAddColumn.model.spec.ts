import { TableAddColumnOperation } from './tableAddColumn.model'
import { generateSchemaAndValidate } from '../../../../utilities'

describe('Operation Model: TableAddColumn Tests', () => {
  let schema
  beforeAll(() => {
    const response = generateSchemaAndValidate(TableAddColumnOperation)
    schema = response.schema
  })

  it('should match this JSON Schema', () => {
    expect(schema.properties).toMatchSnapshot()
  })
})
