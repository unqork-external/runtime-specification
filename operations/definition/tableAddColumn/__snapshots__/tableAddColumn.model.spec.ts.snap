// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Operation Model: TableAddColumn Tests should match this JSON Schema 1`] = `
{
  "creatorSummary": {
    "description": "A detailed summary of the operation",
    "type": "string",
  },
  "name": {
    "description": "Name of the operation",
    "type": "string",
  },
  "options": {
    "$ref": "#/definitions/TableAddColumnOperationOptions",
  },
  "type": {
    "const": "TABLE_ADD_COLUMN",
    "examples": [
      "TABLE_ADD_COLUMN",
    ],
    "type": "string",
  },
}
`;
